js/scholarships.js

```diff
@@ -165,4 +178,100 @@
 		$scholarships_container.find('article:visible:odd').css('background-color', '#fff');
 		$scholarships_container.find('article:visible:even').css('background-color', '#eff0f1');
 	});
+
+	var $scholarships_container_top = $scholarships_container.offset().top,
+		$tools = $('.wsuwp-scholarships-tools');
+
+	// Toggle visibility of the back to top button based on scroll position.
+	$(document).on('scroll', function () {
+		if ($(window).scrollTop() >= $scholarships_container_top) {
+			$tools.show();
+		} else {
+			$tools.hide();
+		}
+	});
+
+	// Jump to the top of the page when the back to top button is clicked.
+	// (This is here only to prevent a hash from being appended to the URL.)
+	$('.back-to-top').on('click', function (e) {
+		e.preventDefault();
+
+		$('html, body').scrollTop(0);
+	});
+
+	// Check if storage is available.
+	function storage_available(type) {
+		try {
+			var storage = window[type],
+				x = '__storage_test__';
+			storage.setItem(x, x);
+			storage.removeItem(x);
+			return true;
+		} catch(e) {
+			return false;
+		}
+	}
+
+	// If storage is available and items are stored, display the form field values.
+	if (storage_available('sessionStorage')) {
+		var results = sessionStorage.getItem('results'),
+			filters = JSON.parse(sessionStorage.getItem('filters'));;
+
+		if (sessionStorage.length) {
+			display_values();
+		}
+
+		if (results) {
+			display_results(results);
+		}
+
+		if (filters) {
+			$.each(filters, function () {
+				$('#' + this).trigger('click');
+			});
+		}
+	}
+
+	// Store the form field values.
+	function store_values() {
+		sessionStorage.setItem('age', $('#wsuwp-scholarship-age').val());
```

Comment:
Have you thought about storing all of this as an object rather than individual pieces?

```
var data_to_store = { age: $( "#wsuwp-scholarship-age" ).val(), gpa: ....etc...};
```

It probably doesn't matter too much, but would mean only using `sessionStorage.setItem()` and `sessionStorage.getItem()` once each.